### PR TITLE
RDK-36589: Thunder Service Support for Regionalization

### PR DESCRIPTION
**Pull request by MManja792**

### CODE REVIEW

File: SystemServices/SystemServices.cpp [hunk 1]
```diff
@@ -2216,6 +2218,126 @@ namespace WPEFramework {
...
+			}
+		}
+		catch(...){
+			LOGERR(" Exception caught");
+			return false;
+		}
+		return true;
+	}
+
+	string SystemServices::getDSTTimeZone()
```
Comment (anand-ky) [inline SystemServices/SystemServices.cpp:2329]: Do we need this function. There is already getTimeZoneDSTHelper()

File: SystemServices/SystemServices.cpp [hunk 2]
```diff
@@ -2216,6 +2218,126 @@ namespace WPEFramework {
...
 
+
+	uint32_t SystemServices::setTerritory(const JsonObject& parameters, JsonObject& response)
+	{
+		bool resp = false;
+		if(parameters.HasLabel("territory")){
+			if(!Utils::fileExists(TERRITORYFILE)){
+				LOGWARN(" Territory : Subdirectories created : %d",val );
+			}
+			readTerritoryFromFile();
```
Comment (anand-ky) [inline SystemServices/SystemServices.cpp:2229]: Do we need to read the file? We would just have to write into it.

File: SystemServices/System.json [hunk 3]
```diff
@@ -1807,6 +1821,12 @@
             }
         }
     },
+    "setTeritory":{
```
[the remark this answers is not in the document]
Comment (MManja792) [reply SystemServices/System.json]: Changed

File: SystemServices/SystemServices.cpp (hunk 2 above)
Comment (MManja792) [reply SystemServices/SystemServices.cpp]: we should get the oldTerritory from file

File: SystemServices/System.json [hunk 4]
```diff
@@ -2056,6 +2076,65 @@
...
+			
+		}
+	
+	},
+	"onTimeZoneDSTChanged":{
+		"summary": "Triggered when the Time zone changed",
+		"params":{
+			"type": "object",
+			"properties": {
+				"currentDSTTime":{
```
[the remark this answers is not in the document]
Comment (MManja792) [reply SystemServices/System.json]: Updated

**Review by MManja792 — COMMENTED** [review]
I have updated the changes suggested.

**Review by anand-ky — APPROVED** [review]

**Pull request merged**